3-extra/1-count-words.js

```diff
@@ -25,9 +25,10 @@
 
 function countWords(string) {
   const wordCount = {};
-
-  // write code here
-
+  if (string.length!=0){
+    const wordArray = string.split(" ");
+    wordArray.forEach(function (x) { wordCount[x] = (wordCount[x] || 0) + 1; });
```

Comment:
Nice solution! Can you explain how `wordCount[x] = (wordCount[x] || 0) + 1` works?

Reply:
The wordCount[x] inside bracket is either undefined (undefined is falsy value) or a positive integer number and the sign || is choose the value of   wordCount[x] if it is defined otherwise it use the second argument which is 0 here.